### ISSUE TITLE
Getting error 'The server rejected the request because the client has exceeded request limits.'

### ISSUE THREAD

**Issue by Julie-House** [IEvangelist/resource-translator#46]
Hi,

We're getting this error when running. I had thought the action would only use the changed resources, but it's attempting to do the whole of all our files (we have 6 files for our convenience of finding things). It's managing to translate 19 batches of the first file before hitting this error, translating to 3 different languages. We're using the latest version of the action (2.1.7). It then continues to try and do the rest of the files but we've already exceeded our limits. 

So, firstly, why is it trying to do the whole file instead of just the changes? Secondly, Azure aren't terribly helpful from what i've seen on the list of errors. Is this something that could pass in a matter of minutes? Could we add the ability to slow down to not hit the limit? Is this a daily or hourly limit?

The whole run time is just over a minute. I'd have no problem if i could set a variable to build in a slow down if that's what it takes!

Thanks

Julie

**Comment by IEvangelist** [comment]
Hi @Julie-House,

I have no idea how I missed this issue. Are you still trying to use this GitHub Action? I've been stretched thin and haven't had time to work on side projects lately, but ideally this would be rate limit aware.

**Comment by Julie-House** [comment]
Hi @IEvangelist !

I would really like to use this action. It would help us immensely. As it happens, our desire to go multi-lingual was put on hold so I never had to find a solution, but I was thinking I would need to fork and fix this (if I could figure it out!). We do plan on moving forward with this in the version we have just started though, so if you can come up with something in the next couple of months, that would be awesome!

**Comment by IEvangelist** [comment]
Hi again, @Julie-House - are you using the free Azure Cognitive Services API? There are different rate limits in place: [Request limits for Translator](https://learn.microsoft.com/en-us/azure/cognitive-services/translator/request-limits).

I could potentially expose a consumer-defined setting, that shares their service tier, with that we could then employ a rate-limit aware mechanism that aligns with those rules. It could be an optional setting to avoid breaking any existing users. It could default to `free` tier, and consider error codes, latency, and request limits.

**Comment by Julie-House** [comment]
@IEvangelist ,

yes, we're using the free version. I'm not sure if we might have scope to upgrade, would like to think maybe we could. But as with all things, until it's out and being used by customers, they don't want the upfront costs. 

Choosing a tier would be a suitable thing to do.

The main thing was that it seemed to be translating all the files instead of only the few changes that may have been made. We've got the initial set of translations, I just want this to run after each PR (or maybe once a day), and if changes are detected in the resx files, to only run for the new resource values. Once we start getting someone verifying and updating the foreign files we don't necessarily want those to be automatically regenerated. 

It's been a long time since I looked at this too! I'll do a bit of digging to see where I got up to and what I felt was outstanding and see what other info I can provide you with.
